cmake/compiler.cmake

```diff
@@ -139,8 +139,14 @@ find_library(UNWIND_LIBRARY PATH_SUFFIXES system NAMES ${UNWIND_LIB_NAME})
 
 # Disabled backtraces support on FreeBSD by default, because of
 # gh-4278.
+# Disabled backtraces support on arm64 by default, because of
+# gh-6272, gh-6060 and gh-6222.
 set(ENABLE_BACKTRACE_DEFAULT OFF)
-if (NOT TARGET_OS_FREEBSD AND UNWIND_LIBRARY AND HAVE_LIBUNWIND_H)
+if (NOT TARGET_OS_FREEBSD AND NOT CMAKE_SYSTEM_PROCESSOR MATCHES "arm*" AND
+    NOT CMAKE_SYSTEM_PROCESSOR MATCHES "aarch*" AND
+    NOT CMAKE_HOST_SYSTEM_PROCESSOR MATCHES "arm*" AND
```

Comment:
Why do you use both `CMAKE_SYSTEM_PROCESSOR` and `CMAKE_HOST_SYSTEM_PROCESSOR`?

Reply:
Because they are not always the same thing, for instance, on our M1 CI runner.
`CMAKE_HOST_SYSTEM_PROCESSOR` is same as the `uname -m` while
`CMAKE_SYSTEM_PROCESSOR` holds value `x86_64`.
But, in case of cross-compiling `CMAKE_HOST_SYSTEM_PROCESSOR` usually doesn't hold the value we need while `CMAKE_SYSTEM_PROCESSOR` does.  
In any case when at least one of the variables corresponds to the Arm architecture, backtraces aren't guaranteed to work correctly.

Reply:
Got it, thanks!

P.S. should we update github-ci build for arm? I suppose within this commit we should remove CMake parameters which force on build with disabled backtrace for arm64.